app/src/main/java/fr/free/nrw/commons/achievements/AchievementsActivity.java

```diff
@@ -0,0 +1,124 @@
+package fr.free.nrw.commons.achievements;
+
+import android.annotation.SuppressLint;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.os.Environment;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.DisplayMetrics;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.RelativeLayout;
+import android.widget.Toast;
+import android.widget.Toolbar;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Random;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import dagger.Binds;
+import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.theme.NavigationBaseActivity;
+
+public class AchievementsActivity extends NavigationBaseActivity {
+
```

Comment:
Please write Javadocs for the class as well (not just the methods). You can briefly describe what AchievementsActivity does.